chainer/training/extensions/evaluator.py

```diff
@@ -88,6 +91,22 @@ def __init__(self, iterator, target, converter=convert.concat_examples,
         self.eval_hook = eval_hook
         self.eval_func = eval_func
 
+        if chainer.is_debug():
+            for key, iter in six.iteritems(iterator):
+                if (isinstance(iter, (iterators.SerialIterator,
+                                      iterators.MultiprocessIterator,
+                                      iterators.MultithreadIterator)) and
+                        getattr(iter, 'repeat', False)):
+                    msg = 'The repeat property of the iterator {} '
+                    'is set to True. Typically, evaluator sweeps '
+                    'over iterators until it raises StopIteration. '
+                    'But this iterator might not raise the exception, '
+                    'which means the evaluation could'
+                    'go into an infinite loop in evaluation. '
+                    'We recommend to check the configuration '
+                    'of iterators'.format(key)
+                    warnings.warn(msg)
```

Comment:
I feel this message is a bit like implementation detail, and relationship between `repeat` property and the behavior is not clearly stated.

How about something like this?

*The repeat property of the iterator {} is set to True. Typically, evaluator sweeps over iterators until it stops iteration, but as its repeat property is True, this iterator might not stop and the evaluator could go into an infinite loop. We recommend to check the configuration of the iterator.*

Reply:
Fixed.